### DISCUSSION TITLE
ojs_define() in Quarto Book to define variable in all chapters

### DISCUSSION THREAD

**Discussion by rgaiacs** [quarto-dev/quarto-cli#3587] [Q&A] [answered]
Related to https://github.com/quarto-dev/quarto-cli/issues/3568

[Zip file with minimal working example](https://github.com/quarto-dev/quarto-cli/files/10172663/ojs-book.zip)

Consider a project that has one chapter as

``````
# Summary

```{ojs}
1 + 1
```

Key fact is ${key_fact}
``````

and another chapter as

``````
# Methods

```{r}
key_fact <- 1 + 1
ojs_define(key_fact = key_fact)
```
``````

At the moment, Observable's `${key_fact}` silent fails because the variable is only defined in one file.

![Screenshot 2022-12-07 at 11-59-02 Minimal Working Example](https://user-images.githubusercontent.com/1506457/206085190-ff3ee955-4555-4a0f-ab7e-d2253d6b155b.png)

Will be great if Quarto allows the option to globally define the variable across a book.

**Comment by cscheid** [comment, marked as answer]
Use imports across pages instead:


## foo.qmd

````
```{ojs}
x = 1
```
````

## bar.qmd

````
```{ojs}
import { x } from "./foo.qmd";
```
````

That should work.

**Reply by rgaiacs** [reply]
Thanks! Works for me.

**Comment by mcanouil** [comment]
(Changing this to "Q&A")
